src/main/java/com/fasterxml/jackson/databind/DeserializationContext.java

```diff
@@ -468,8 +469,14 @@ public final Object findInjectableValue(Object valueId,
         throws JsonMappingException
     {
         if (_injectableValues == null) {
-            return reportBadDefinition(ClassUtil.classOf(valueId), String.format(
-"No 'injectableValues' configured, cannot inject value with id [%s]", valueId));
+            final JacksonInject.Value injectableValue = getAnnotationIntrospector()
+                    .findInjectableValue(forProperty.getMember());
+
+            return injectableValue.getOptional() || !isEnabled(DeserializationFeature.FAIL_ON_UNKNOWN_INJECT_VALUE)
```

Comment:
I think logic here is incorrect. Basically, default would be state of `DeserializationFeature.FAIL_ON_UNKNOWN_INJECT_VALUE`, but overridden by `injectableValue.getOptional()` if (and only if) it is not `null` (null is for `DEFAULT`).
I can change this.